librz/analysis/class.c

```diff
@@ -478,13 +497,78 @@ static RzAnalysisClassErr rz_analysis_class_add_attr_unique(RzAnalysis *analysis
 // Format: addr,vtable_offset
 
 static char *flagname_method(const char *class_name, const char *meth_name) {
+	if (rz_str_startswith(meth_name, "method.")) {
+		return rz_str_new(meth_name);
+	}
 	return flagname_attr("method", class_name, meth_name);
 }
 
 RZ_API void rz_analysis_class_method_fini(RzAnalysisMethod *meth) {
 	free(meth->name);
 }
 
+RZ_API void rz_analysis_class_method_recover(RzAnalysis *analysis, RzBinClass *class, RzList *methods) {
+	RzListIter *iter_method;
+	RzBinSymbol *sym;
+	rz_list_foreach (methods, iter_method, sym) {
+		if (!rz_analysis_class_method_exists(analysis, class->name, sym->name)) {
+			//detect constructor or destructor but not implemented
+			//Temporarily set to default
+			RzAnalysisMethod method;
+			method.addr = sym->vaddr;
+			method.vtable_offset = -1;
+			RzAnalysisFunction *fcn = rz_analysis_get_function_at(analysis, sym->vaddr);
+			method.name = fcn ? rz_str_new(fcn->name) : rz_str_new(sym->name);
+			method.method_type = RZ_ANALYSIS_CLASS_METHOD_DEFAULT;
+			rz_analysis_class_method_set(analysis, class->name, &method);
+			rz_analysis_class_method_fini(&method);
+		}
+	}
+}
+
+RZ_API bool rz_analysis_class_method_exists(RzAnalysis *analysis, const char *class_name, const char *meth_name) {
+	char *content = rz_analysis_class_get_attr(analysis, class_name, RZ_ANALYSIS_CLASS_ATTR_TYPE_METHOD, meth_name, false);
+	if (!content) {
+		return false;
+	}
+
+	if (!rz_str_sanitize_sdb_key(meth_name)) {
```

Comment:
Why this check here? I don't think it is related to checking if a method exists. Moreover, the returned pointer is leaked.

Reply:
Sorry, I copied it from `rz_analysis_class_method_get()` but I forgot to check what it does..